src/rust/ide/src/controller/module.rs

```diff
@@ -42,7 +52,7 @@ use shapely::shared;
 pub struct Metadata {
     /// Metadata used within ide.
     #[serde(default="default")]
-    pub ide : serde_json::Value,
+    pub ide : HashMap<ast::ID,NodeMetadata>,
```

Comment:
I'd consider renaming `ide` to `nodes`. Or otherwise introduce struct, because ide might want to store more than just id->nodemetadata.